test/framework/envtest/cluster/provisioner_k3d.go

```diff
@@ -0,0 +1,239 @@
+// +build test
+
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rancher/k3d/v3/pkg/cluster"
+	"github.com/rancher/k3d/v3/pkg/runtimes"
+	"github.com/rancher/k3d/v3/pkg/types"
+
+	"github.com/rancher/octopus/test/framework/finder"
+)
+
+// ProvisionerTypeK3d indicates the testing cluster is created by k3d.
+const ProvisionerTypeK3d ProvisionerType = "k3d"
+
+// NewK3d creates a github.com/rancher/k3d cluster.
+func NewK3d() (*K3d, error) {
+	var k3d = K3d{}
+	if err := finder.Parse(&k3d); err != nil {
+		return nil, errors.Wrap(err, "failed to parse K3d cluster argument from envs")
+	}
+	return &k3d, nil
+}
+
+type K3d struct {
+	// Specify the image for running cluster,
+	// configure in "K3S_IMAGE" env,
+	// default is "rancher/k3s:v1.18.2-k3s1".
+	Image string `env:"name=K3S_IMAGE,default=rancher/k3s:v1.18.2-k3s1"`
+
+	// Specify the name of cluster,
+	// configure in "K3S_CLUSTER_NAME" env,
+	// default is "edge".
+	ClusterName string `env:"name=K3S_CLUSTER_NAME,default=edge"`
+
+	// Specify the amount of control-plane nodes,
+	// configure in "K3S_CONTROL_PLANES" env,
+	// default is "1".
+	ControlPlanes int `env:"name=K3S_CONTROL_PLANES,default=1"`
+
+	// Specify the amount of worker nodes,
+	// configure in "K3S_WORKERS" env,
+	// default is "3".
+	Workers int `env:"name=K3S_WORKERS,default=3"`
+
+	// Specify the wait timeout for bringing up cluster,
+	// configure in "K3S_WAIT_TIMEOUT" env,
+	// default is "10m".
+	WaitTimeout time.Duration `env:"name=K3S_WAIT_TIMEOUT,default=10m"`
```

Comment:
10m is a bit too long, 5 minutes should be sufficient enough since it runs in cloud machines(low network latency).

Reply:
Updated